apps/gitness/src/user-management/user-management-container.tsx

```diff
@@ -0,0 +1,112 @@
+import { SandboxSettingsUserManagementPage, useCommonFilter } from '@harnessio/playground'
+import {
+  useAdminListUsersQuery,
+  useAdminUpdateUserMutation,
+  useAdminDeleteUserMutation,
+  useUpdateUserAdminMutation,
+  AdminListUsersQueryQueryParams
+} from '@harnessio/code-service-client'
+import { useQueryClient } from '@tanstack/react-query'
+import { parseAsInteger, useQueryState } from 'nuqs'
+import { PageResponseHeader } from '../types'
+
+export const UserManagementPageContainer = () => {
+  const queryClient = useQueryClient()
+
+  const { query: _currentQuery, sort } = useCommonFilter<AdminListUsersQueryQueryParams['sort']>()
+  // const [query, _] = useQueryState('query', { defaultValue: currentQuery || '' })
+  const [page, setPage] = useQueryState('page', parseAsInteger.withDefault(1))
+  const changePage = (pageNum: number) => setPage(pageNum)
```

Comment:
how is this different from just calling `setPage` directly wherever needed?

Reply:
Since we're passing it down as a prop to the child component, I think passing a callback rather than the `setState` itself is a better pattern.